FluidNC/src/Maslow/Maslow.cpp

```diff
@@ -679,76 +635,72 @@ if(decompress){
-//   int distY = currentYTarget - toY;
-
-//   if(distX > 0 )  xDirection = -1;
-//   else if(distX < 0) xDirection = 1;
-//   else xDirection = 0;
-
-//   if(distY > 0 )  yDirection = -1;
-//   else if(distY < 0) yDirection = 1;
-//   else yDirection = 0;
 
   int xDirection = currentXTarget - toX > 0 ? -1 : 1;
   int yDirection = currentYTarget - toY > 0 ? -1 : 1;
-  
-  int comply_spd = 500; //doesn't actually do anything now
-  
+  if(abs(currentXTarget - toX) <= stepSize ) xDirection = 0; 
+  if(abs(currentYTarget - toY) <= stepSize ) yDirection = 0; 
+
   switch (direction) {
     case UP:
-        setTargets(currentXTarget + xDirection * stepSize, currentYTarget + yDirection * stepSize, 0, true, true, false, false);
-        axisTL.recomputePID();
-        axisTR.recomputePID();
-        axisBL.comply(comply_spd);
-        axisBR.comply(comply_spd);
-        if( onTarget(toX, toY, getTargetX(), getTargetY(), 0.25) ) {
+        setTargets(toX,toY, 0, true, true, false, false);
+        axisTL.recomputePID(500);
+        axisTR.recomputePID(500);
+        axisBL.comply();
+        axisBR.comply();
+        //if( onTarget(toX, toY, getTargetX(), getTargetY(), 0.25) ) { // this doesn't check if we actually got to the position with belts
+        if(axisTL.onTarget(0.25) && axisTR.onTarget(0.25)){
             stopMotors();
             reset_all_axis();
             decompress = true; //Reset for the next pass
             return true;
         }
+        //}
         break;
       case DOWN:
-            setTargets(currentXTarget + xDirection * stepSize, currentYTarget + yDirection * stepSize, 0, false, false, true, true);
-            axisTL.comply(comply_spd);
-            axisTR.comply(comply_spd);
-            axisBL.recomputePID();
-            axisBR.recomputePID();
-            if( onTarget(toX, toY, getTargetX(), getTargetY(), 0.25) )  {
+            setTargets(toX,toY, 0, false, false, true, true);
+            axisTL.comply();
+            axisTR.comply();
+            axisBL.recomputePID(500);
+            axisBR.recomputePID(500);
+            //if( onTarget(toX, toY, getTargetX(), getTargetY(), 0.25) )  {
+            if(axisBL.onTarget(0.25) && axisBR.onTarget(0.25)){
                 stopMotors();
                 reset_all_axis();
                 decompress = true; //Reset for the next pass
                 return true;
             }
+            //}
             break;
       case LEFT:
-            setTargets(currentXTarget + xDirection * stepSize, currentYTarget + yDirection * stepSize, 0, true, false, true, false);
-            axisTL.recomputePID();
-            axisTR.comply(comply_spd);
-            axisBL.recomputePID();
-            axisBR.comply(comply_spd);
-            if( onTarget(toX, toY, getTargetX(), getTargetY(), 0.25) ) {
+            setTargets(toX,toY, 0, true, false, true, false);
+            axisTL.recomputePID(500);
+            axisTR.comply();
+            axisBL.recomputePID(500);
```

Comment:
Is the 500 here to limit the speed? Is that limiting the maximum output of the PID controller on the 0-1023 scale?